test/functional/users_controller_test.rb

```diff
@@ -364,6 +365,39 @@ class UsersControllerTest < ActionController::TestCase
     assert_response :unprocessable_entity
   end
 
+  context "an active session is present" do
+    test "#login redirects to root(dashboard)" do
+      get :login, set_session_user
+
+      get :login
+      assert_redirected_to root_path
+    end
+
+    test "for user :one -> #login redirects to root(dashboard)" do
```

Comment:
Sorry, I missed this point. It's worth leaving out unless there's a difference in the tested code for the two types of user, just to minimise the test time.